### PR TITLE
Gateway/singularity interaction fix

### LINKED ISSUE #13325: If the singularity eats the gateway, you can still warp to it
And you end up in nullspace

### PR DESCRIPTION
**Pull request by imtakingabreakdontatme**
Fixes #13325

> but kor why didn't you put the hard_del in /machinery/gateway and let centerstation/centeraway inherit it?

I am under the impression that we're not supposed to hard delete things if at all possible, and there is no need to hard del the other 8 parts of each gate, just the two center pieces.

If I'm wrong on that count it's an easy fix

### CODE REVIEW

**Comment by Jordie0608** [comment]
Why not just fix whatever is preventing it from qdeling?

**Comment by KorPhaeron** [comment]
@Jordie0608 

Getting stuck in the singulo billion object queue.

Unless I completely misunderstand how singulo consuming objects works.

**Comment by AnturK** [comment]
Just check if other side is in the qdel queue on teleporting ? qdeleted(A) is the proc.

**Comment by KazeEspada** [comment]
Boring. It should of blown up.

File: code/modules/awaymissions/gateway.dm [hunk 1]
```diff
@@ -154,6 +154,9 @@ var/obj/machinery/gateway/centerstation/the_gateway = null
 		user << "\black The gate is already calibrated, there is no work for you to do here."
 		return
 
+/obj/machinery/gateway/centerstation/Destroy()
```
Comment (MatthewWork) [inline code/modules/awaymissions/gateway.dm]: You left the awful type pathing in. centerstation and centeraway have so much duped code. It's awful.

**Comment by Jordie0608** [comment]
@KorPhaeron An atom just needs to have all references to it cleared before it'll qdel

**Comment by tkdrg** [comment]
yeah just add a qdeleted() check to the telport code instead

**Comment by KorPhaeron** [comment]
@tkdrg 

Done

**Comment by KorPhaeron** [comment]
Or not

File: code/modules/awaymissions/gateway.dm [hunk 2]
```diff
@@ -232,8 +236,12 @@ var/obj/machinery/gateway/centerstation/the_gateway = null
 
 
 /obj/machinery/gateway/centeraway/Bumped(atom/movable/M as mob|obj)
-	if(!ready)	return
-	if(!active)	return
+	if(!ready)	
+		return
+	if(!active)	
+		return
+	if(!stationgate || qdeleted(station))
```
Comment (tkdrg) [inline code/modules/awaymissions/gateway.dm]: the second one of these needs to be `stationgate`

**Comment by KorPhaeron** [comment]
@tkdrg 

Thanks. Should be ready now

**Pull request merged**